### PR TITLE
Feature: Add page anchors (inter page links)

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: #299
* Resolves: #2173
* Supersedes: #2520
* Target version: master 

### Summary
This is a replacement of #2520 (pushed on this repository to enable the CI correctly).

Most parts were already resolved by #2688 (adding IDs to headings, creating the slugs etc), so this just provides the missing user interface (clickable anchor, copyable link, scrolling into view).

It shows the permalink when hovering the heading:

Showing the permalink on bright mode
Showing the permalink on dark mode

### CODE REVIEW

**Review by vinicius73 — APPROVED** [review]

**Comment by vinicius73** [comment]
I just miss some testing coverage.

**Comment by susnux** [comment]
> I just miss some testing coverage.

Ok then I will add some tests for it, I guess checking for
* [ ] anchors to be available, and
* [ ] clicking on links scrolls the heading into view

is sufficient?

File: src/nodes/Heading/HeadingView.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+<template>
+	<NodeViewWrapper ref="content"
+		:as="domElement"
+		v-bind="node.attrs"
+		tabindex="-1">
+		<a aria-hidden="true"
+			class="anchor-link"
+			:href="href"
+			:title="t('text', 'Permalink')"
```
Comment (max-nextcloud) [inline src/nodes/Heading/HeadingView.vue:29]: I'm not sure Permalink is a good term here. For me Permalink means this link will stay available over time. However if the section or the file are renamed this link might change - obviously depending on the url used initially.

First ideas that come to my mind are 'Link to this section', 'Link here' or 'Jump here'.

**Review by max-nextcloud — COMMENTED** [review]
Looks great. Thanks for taking care of this. Just a minor wording comment.

File: src/nodes/Heading/HeadingView.vue (hunk 1 above)
Comment (susnux) [reply src/nodes/Heading/HeadingView.vue]: I reword it to make it more meaningful.

**Comment by susnux** [comment]
Tasks done @vinicius73 @max-nextcloud

**Comment by max-nextcloud** [comment]
Wording change looks good. Thanks!

**Comment by vinicius73** [comment]
It has style issues

image

**Comment by susnux** [comment]
> It has style issues

Which browser do you use? (So I can reproduce and try to fix it).

**Comment by vinicius73** [comment]
> Which browser do you use? (So I can reproduce and try to fix it).

Chrome 104

I've sent a commit about this, check if it is ok pls.

**Comment by susnux** [comment]
@vinicius73 thank you, now it works perfectly!

**Comment by juliushaertl** [comment]
Very nice. Thanks again for all the great contributions @susnux 

In case you haven't seen we will have the Nextcloud conference taking place in about a moth again in Berlin with a lot of talks over the weekend but also some coding days afterwards and it would be quite cool to have you there and meet each other in person. There is also travel support for contributors. More info is available at http://nextcloud.com/conference-2022/ or if you have any questions just let me know.

**Comment by susnux** [comment]
@juliushaertl I would loved to, but besides being invited for a important familiar event, I am down with the flu so it is and was not possible for me participate. Maybe in the future!
